CoreCLR failed when build with MSVC on Windows

Hello,

I tried to build CoreCLR on Windows with VS2017 Update2. It failed to build due to the error CS2001: Source file 'D:\CoreCLR\src\bin\obj\Windows_NT.x86.Release\System.Private.CoreLib\..\eventing\DotNETRuntimeEventSource.cs' could not be found. This issue is caused by master branch revision [0fbd0f5](https://github.com/dotnet/coreclr/commit/0fbd0f535e566bcde591701adc2de51ee0f020af). Could you please help take a look at this? Thank you!

**You can repro this issue as the steps below:**
1. git clone https://github.com/dotnet/coreclr D:\CoreCLR\src
2. Open a clean x86 prompt (C:\windows\syswow64\cmd.exe) and browse to D:\CoreCLR\src
3. build x86 release -Rebuild -- /clp:ShowCommandLine

**Error info:**
CSC : error CS2001: Source file 'D:\CoreCLR\src\bin\obj\Windows_NT.x86.Release\System.Private.CoreLib\..\eventing\DotNETRuntimeEventSource.cs' could not be found. [D:\CoreCLR\src\src\System.Private.CoreLib\System.Private.CoreLib.csproj]

--- a/Documentation/building/cross-building.md
+++ b/Documentation/building/cross-building.md
@@ -79,11 +79,11 @@ Once the rootfs has been generated, it will be possible to cross compile CoreCLR
 
 So, without `ROOTFS_DIR`:
 
-    ben@ubuntu ~/git/coreclr/ $ ./build.sh arm debug verbose cross -rebuild
+    ben@ubuntu ~/git/coreclr/ $ ./build.sh arm debug verbose cross
 
 And with:
 
-    ben@ubuntu ~/git/coreclr/ $ ROOTFS_DIR=/home/ben/coreclr-cross/arm ./build.sh arm debug verbose cross -rebuild
+    ben@ubuntu ~/git/coreclr/ $ ROOTFS_DIR=/home/ben/coreclr-cross/arm ./build.sh arm debug verbose cross
 
 As usual the resulting binaries will be found in `bin/Product/BuildOS.BuildArch.BuildType/`
 
@@ -115,7 +115,7 @@ The following instructions assume you are on a Linux machine such as Ubuntu 14.0
 To build System.Private.CoreLib for Linux, run the following command:
 
 ```
-    lgs@ubuntu ~/git/coreclr/ $ build.sh arm debug verbose -rebuild
+    lgs@ubuntu ~/git/coreclr/ $ build.sh arm debug verbose
 ```
 
 The output is at bin/Product/<BuildOS>.arm.Debug/System.Private.CoreLib.dll.

--- a/Documentation/building/windows-instructions.md
+++ b/Documentation/building/windows-instructions.md
@@ -144,7 +144,6 @@ Build has a number of options that you can learn about using build -?.   Some of
 
  * -skiptests - don't build the tests.   This can shorten build times quite a bit, but means you can't run tests.
  * -release - build the 'Release' build type that does not have extra development-time checking compiled in.
- * -rebuild - force the build not to be incremental but to recompile everything.   
  You want this if you are going to do performance testing on your build. 
 
 See [Using Your Build](../workflow/UsingYourBuild.md) for instructions on running code with your build.  

--- a/build.cmd
+++ b/build.cmd
@@ -135,6 +135,17 @@ if /i "%1" == "release"             (set __BuildTypeRelease=1&set processedArgs=
 
 if /i "%1" == "-priority"           (set __Priority=%2&shift&set processedArgs=!processedArgs! %1=%2&shift&goto Arg_Loop)
 
+REM Explicitly block -Rebuild.
+if /i "%1" == "Rebuild" (
+    echo "ERROR: 'Rebuild' is not supported.  Please remove it."
+    goto Usage
+)
+if /i "%1" == "-Rebuild" (
+    echo "ERROR: 'Rebuild' is not supported.  Please remove it."
+    goto Usage
+)
+
+
 REM All arguments after this point will be passed through directly to build.cmd on nested invocations
 REM using the "all" argument, and must be added to the __PassThroughArgs variable.
 if [!__PassThroughArgs!]==[] (
@@ -890,7 +901,6 @@ echo -skiprestore: skip restoring packages ^(default: packages are restored duri
 echo -disableoss: Disable Open Source Signing for System.Private.CoreLib.
 echo -priority=^<N^> : specify a set of test that will be built and run, with priority N.
 echo -officialbuildid=^<ID^>: specify the official build ID to be used by this build.
-echo -Rebuild: passes /t:rebuild to the build projects.
 echo -crossgenaltjit ^<JIT dll^>: run crossgen using specified altjit ^(used for JIT testing^).
 echo portable : build for portable RID.
 echo.

--- a/build.sh
+++ b/build.sh
@@ -49,7 +49,6 @@ usage()
     echo "-skiprestore: skip restoring packages ^(default: packages are restored during build^)."
     echo "-disableoss: Disable Open Source Signing for System.Private.CoreLib."
     echo "-officialbuildid=^<ID^>: specify the official build ID to be used by this build."
-    echo "-Rebuild: passes /t:rebuild to the build projects."
     echo "-stripSymbols - Optional argument to strip native symbols during the build."
     echo "-skipgenerateversion - disable version generation even if MSBuild is supported."
     echo "-ignorewarnings - do not treat warnings as errors"
@@ -880,6 +879,10 @@ while :; do
               echo "ERROR: 'osgroup' requires a non-empty option argument"
               exit 1
             fi
+            ;;
+        rebuild|-rebuild)
+            echo "ERROR: 'Rebuild' is not supported.  Please remove it."
+            exit 1
             ;;
 
         *)

--- a/config.json
+++ b/config.json
@@ -282,12 +282,6 @@
       "values": [ "false", "true"],
       "defaultValue": "true"
     },
-    "Rebuild": {
-      "description": "Specifies rebuild target.",
-      "valueType": "target",
-      "values": [],
-      "defaultValue": ""
-    },
     "MsBuildLog": {
       "description": "Specifies build logs for build alias.",
       "valueType": "passThrough",

--- a/tests/scripts/arm32_ci_script.sh
+++ b/tests/scripts/arm32_ci_script.sh
@@ -227,7 +227,7 @@ function cross_build_coreclr {
     fi
 
     #Cross building for emulator rootfs
-    ROOTFS_DIR="$__ARMEmulRootfs" CPLUS_INCLUDE_PATH=$LINUX_ARM_INCPATH CXXFLAGS=$LINUX_ARM_CXXFLAGS ./build.sh $__buildArch cross $__verboseFlag $__skipMscorlib clang3.5 $__buildConfig -rebuild
+    ROOTFS_DIR="$__ARMEmulRootfs" CPLUS_INCLUDE_PATH=$LINUX_ARM_INCPATH CXXFLAGS=$LINUX_ARM_CXXFLAGS ./build.sh $__buildArch cross $__verboseFlag $__skipMscorlib clang3.5 $__buildConfig
 
     #Reset the code to the upstream version
     (set +x; echo 'Rewinding HEAD to master code')

